public/app/plugins/datasource/zipkin/datasource.ts

```diff
@@ -35,31 +35,29 @@ export class ZipkinDatasource extends DataSourceApi<ZipkinQuery> {
     }
   }
 
-  async metadataRequest(url: string, params?: Record<string, any>): Promise<any> {
+  async metadataRequest(url: string, params?: Record<string, any>) {
     const res = await this.request(url, params, { hideFromInspector: true }).toPromise();
     return res.data;
   }
 
-  async testDatasource(): Promise<any> {
+  async testDatasource() {
     await this.metadataRequest(`${apiPrefix}/services`);
-    return true;
+    return { status: 'success', message: 'Data source is working' };
   }
 
   getQueryDisplayText(query: ZipkinQuery) {
     return query.query;
   }
 
-  private request<T = any>(apiUrl: string, data?: any, options?: Partial<BackendSrvRequest>): Observable<{ data: T }> {
-    // Hack for proxying metadata requests
-    const baseUrl = `/api/datasources/proxy/${this.instanceSettings.id}`;
+  private request<T = any>(apiUrl: string, data?: any, options?: Partial<BackendSrvRequest>) {
     const params = data ? serializeParams(data) : '';
-    const url = `${baseUrl}${apiUrl}${params.length ? `?${params}` : ''}`;
+    const url = `${this.instanceSettings.url}${apiUrl}${params.length ? `?${params}` : ''}`;
```

Comment:
Isn't this basically changing the access mode from proxy to direct? Before we were using the data source proxy with the `/api/datasources/proxy` but `this.instanceSettings.url` should contain direct url for the datasource.

Reply:
Nah. I was trying to figure out how but the url is the datasource proxy see 
<img width="1422" alt="Screenshot 2021-02-04 at 13 19 31" src="https://user-images.githubusercontent.com/13729989/106892098-e3323c00-66eb-11eb-914f-787ed6a9ab09.png">

Reply:
Ah interesting didn't know we swap that for the proxy one